rfcs/react-components/convergence/semantic-shared-colors.md

```diff
@@ -0,0 +1,161 @@
+- 👍 Designers can "freely" change the status colors' mapping.
+- 👍 Partners can override persona and status colors independently.
+- 👎 We will remove some "basic" colors like (slots of) `red`, `green`, `yellow` - are we sure those have been used as semantic colors only?
+- 👎 1A: adds ~70 new tokens
+- 👎 1A: can cause temporarily inconsistent UI colors between library and partners
+- 👍 1B: we can use the current mapping in v1 theme and the new mapping in v2 theme - partners can control the moment when they switch to the new mapping
+- 👎 1B: requires us maintaining the double amount of themes
+- 👎 1C: breaks partners (the questions is how much, but...)
+
+### 2. Continue with the current approach
+
+The current names are set for v9, let's live with them 🤷‍♂️.
+
+Add "status" slots to `cranberry`, add `orange` as a status color.
+
+Update the sources in the library to use the new colors. Ask partners to update their sources.
+
+#### Pros and Cons
+
+- 👍 Minimal change, no new patterns, just adds 15 new tokens.
+- 👎 Can cause temporarily inconsistent UI colors between library and partners.
+- 👎 Not improving designers' workflow.
+
+### 3. Hack the hex values
+
+The current names are set for v9, let's live with them 🤷‍♂️.
+
+Change the hex values for `red` to `cranberry` values, `yellow` to `orange`.
+
+#### Pros and Cons
+
+- 👍 Change does not affect components or partner code.
+- 👍 No new tokens.
+- 👎 Designers will scream as "Yellow and orange are two different colors!".
+- 👎 There will be `red` and `cranberry` persona colors in the theme, but they will have the same hex values (but sure, we can take `cranberry` and change its hex values to, let's say `red` 🤯).
+
+### 4. Add semantic color tokens mapped to the current shared colors
+
+All tokens are mapped to CSS variables with the same name: `colorPaletteRedBackground1` token is mapped to `--colorPaletteRedBackground1` CSS variable. From the very first release of FUIR9, we [always warned partners](https://react.fluentui.dev/?path=/docs/concepts-developer-theming--page#do-not-use-css-variables-directly) that they should never depend on this fact:
+
+> Do not use CSS variables directly
+>
+> ⚠ Never use theme CSS variables directly! The CSS variables implementation of the theme is internal to the library. We might eventually decide to change the variable names, hash them or even use direct values instead of some variables. Always use the tokens to access the theme.
+
+Add "status" slots to `cranberry`, add `orange` as a status color.
+
+Add semantic color tokens which are mapped to the current shared colors. For example, `colorPaletteDangerBackground1` is mapped to `--colorPaletteRedBackground1`. `Danger` is still mapped to the old `red` color, but the token name is now semantic. Still not changed to `cranberry`.
+
+Update the sources in the library to use the new colors.
+
+Update the mapping from old colors to the new colors.
```

Comment:
Summarizing a suggestion made offline:
How about instead using the new token names ("danger", etc.) as css variables, and mapping the old (deprecated) token names ("red", etc.) to the new css variables:
```
/** @deprecated Use colorPaletteDangerBackground1 or colorPersonaRedBackground1 */
tokens.colorPaletteRedBackground1 = 'var(--colorPaletteDangerBackground1)'
```

This should mitigate/solve the first two cons of option 4:
* 👍 It's possible to override danger separately from persona-red.
* 👍 The css variable names map to the new, recommended token names for easier debugging.

One additional con of this:
* 👎 If anyone is using the css variable names directly, they will break when we rename them to the new names.

Reply:
> 👍 The css variable names map to the new, recommended token names for easier debugging.

This makes sense, we should optimize debugging for the recommended use case.

My question would be if there would be a future for `colorPaletteRed` as a token. If the intention is for users to stop using it and only use `colorPaletteDanger` (i.e. remove this token on next major), then @behowell's suggestion makes sense.

If we still expect both tokens to be used past the next major, I don't think any kind of mappings in the FUI theme will be scalable because it overriding one token can affect other tokens. This would apply to any mapped tokens we intend to introduce.

Reply:
I'm in favour of both @behowell idea and `1B: "deprecate" the tokens and create a V2 theme which does not contain them - let applications switch when they are ready`.

that v2 gradual approach is something we will be adopting in "near-ish future" so would nicely align. also it would mitigate "unexpected" breaking changes.

Reply:
Added as option 4B.

It makes sense for colors which we are 'deprecating' - `green` and `yellow` should not be used anywhere so we introduce `success` and for backwards compatibility we map `green` to the new `success`.

> 👍 It's possible to override danger separately from persona-red.

Huh, actually I had ignored the fact that in option 4 we would have a different shape of theme and tokens. Added to **Open Issues**, will look into it next week.

> that v2 gradual approach is something we will be adopting in "near-ish future" so would nicely align.

Realized the same problem here - introducing v2 theme would also require introducing v2 tokens which would require an import change in all `.styles.ts` files in the partner codebase :/ What would be our solution? Codemod?

Reply:
> My question would be if there would be a future for colorPaletteRed as a token. If the intention is for users to stop using it and only use colorPaletteDanger (i.e. remove this token on next major), then @behowell's suggestion makes sense and I would vote for that.

This is why I would rather prefer 4 to 4B out of these two options - `red` is still valid.
And I can understand `danger` being mapped to `red` (because the color **is** red). But I do not understand `red` being mapped to `danger` (but I might be biased).

Also designs change - as we see, `danger` mapping is changing from `red` to `cranberry` - semantic colors are less stable. But whatever is red will always be red - I would expect less changes here.